mautic/mautic
# Change pre-commit hook to allow partial commit, but check if there are errors in php-cs-fixer and rector.

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | 🟢 
| New feature/enhancement? (use the a.x branch)      | 🔴
| Deprecations?                          | 🔴
| BC breaks? (use the c.x branch)        | 🔴


## Description
Another take on the issue described in the PR #13185 and then reverted in #13299 . Also added commit from #13078 and refactored to use the new functionality.
/cc @kuzmany @mollux @escopecz 






---
### 📋 Steps to test this PR:


1. Pull down for testing locally.
2. Add some changes to code with errors for both php-cs-fixer and rector.
3. Try to merge other branch *with changes* and see it completed without errors.
4. Try to commit
5. Observe the commit did not succeed due to error in php-cs-fixer.
6. Add changes that will pass the php-cs-fixer.
7. Try to commit.
8. Observe the commit did not succeed now due to error found by rector.
9. Add changes that will pass the rector.
10. Add part of the changes that belong to *one file* to commit, leaving the file partially committed.
11. Observe only changes added to the commit are committed.
